src/main/java/com/uid2/admin/vertx/service/EncryptionKeyService.java

```diff
@@ -548,6 +548,15 @@ private int getOrCreateKeysetId(int siteId)
         Keyset keyset = lookUpKeyset(siteId, currentKeysets);
         if(keyset == null) {
             int newKeysetId = getMaxKeyset(currentKeysets)+1;
+            if(siteId == Const.Data.MasterKeySiteId) {
+                newKeysetId = Const.Data.MasterKeysetId;
```

Comment:
just had a thought - [Migration stage -> keyset.json](https://atlassian.thetradedesk.com/confluence/display/UID2/UID2-545+Sharing+-+Data+Storage+for+Keys+and+Grants#UID2545SharingDataStorageforKeysandGrants-MigrationStage-MultipleKeysets) shows that name should be populated for the special keysets, eg "Master", "Refresh".

Reply:
Added both and  "FallbackPublisher" keyset as well

Reply:
the name are defined in this shared lib PR https://github.com/IABTechLab/uid2-shared/pull/99/files